File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        sort(salary.begin(),salary.end());
        
        double sum=0.0,c=0.0;
        for(int i=1;i<salary.size()-1;++i)
        {
            sum+=salary[i];
            c++;
        }
        return (sum/c);
    }
};

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        double sum = 0d;
        int M = Integer.MIN_VALUE, m = Integer.MAX_VALUE;
        for (int s : salary) {
            m = Math.min(m, s);
            M = Math.max(M, s);
            sum += s;
        }
        return (sum - m - M) / (salary.length - 2);
    }
}